app/src/main/java/com/imashnake/animite/features/media/MediaPageViewModel.kt

```diff
@@ -1,95 +1,93 @@
 package com.imashnake.animite.features.media
 
+import android.graphics.Color
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import com.imashnake.animite.MediaQuery
 import com.imashnake.animite.data.repos.MediaRepository
+import com.imashnake.animite.features.destinations.MediaPageDestination
 import com.imashnake.animite.type.MediaRankType
 import com.imashnake.animite.type.MediaType
 import dagger.hilt.android.lifecycle.HiltViewModel
-import kotlinx.coroutines.Job
 import kotlinx.coroutines.launch
 import java.io.IOException
 import javax.inject.Inject
 
 @HiltViewModel
 class MediaPageViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
     private val mediaRepository: MediaRepository
 ) : ViewModel() {
+    private val navArgs = MediaPageDestination.argsFrom(savedStateHandle)
+
     var uiState by mutableStateOf(MediaUiState())
         private set
 
-    private var fetchJob: Job? = null
-    fun populateMediaPage(id: Int?, mediaType: MediaType) {
-        fetchJob?.cancel()
-        fetchJob = viewModelScope.launch {
+    init {
+        viewModelScope.launch {
             try {
-                val media = mediaRepository.fetchMedia(id, mediaType)
+                val mediaType = MediaType.safeValueOf(navArgs.mediaType)
+                val media = mediaRepository.fetchMedia(navArgs.id, mediaType)
 
-                val score = listOf(Stat(StatLabel.SCORE, media?.averageScore))
+                val score = media?.averageScore?.let { listOf(Stat(StatLabel.SCORE, it)) } ?: emptyList()
 
                 val rankOptions = mapOf(
                     MediaRankType.RATED to StatLabel.RATING,
                     MediaRankType.POPULAR to StatLabel.POPULARITY,
                 )
                 // TODO: This code is a little sus, see if we can create internal models.
-                val ranks = media?.rankings?.filter {
-                    it?.type in rankOptions.keys && it?.allTime == true
-                }?.map {
-                    Stat(rankOptions[it?.type] ?: StatLabel.UNKNOWN, it?.rank)
-                } ?: emptyList()
+                val ranks = media?.rankings
+                    ?.filter { it != null && it.type in rankOptions.keys && it.allTime == true }
+                    ?.map { Stat(rankOptions[it!!.type] ?: StatLabel.UNKNOWN, it.rank) }
+                    ?: emptyList()
 
                 uiState = with(uiState) {
                     copy(
                         bannerImage = media?.bannerImage,
                         coverImage = media?.coverImage?.extraLarge,
-                        color = media?.coverImage?.color,
+                        color = media?.coverImage?.color?.let { Color.parseColor(it) },
```

Comment:
great this is another thing to eliminate

Reply:
Think about how much better these will get with proper null safety